fix : Multivendor-admin Rider username contains empty spaces issue

Nishanth-sebastin opened the pull request:
[Screencast from 27-02-24 08:14:43 AM IST.webm](https://github.com/ninjas-code-official/food-delivery-multivendor/assets/85913314/ae621f1f-60e4-4534-820b-8622e97118ea)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Improved form validation for the rider management system, including detailed error messages.
- **Enhancements**
	- Added username format constraints to disallow spaces for better data integrity.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Nishanth-sebastin commented:
Added validation message for username field:

- Updated validation rules in validation.js to include a new format rule for the username field, ensuring it does not contain spaces.
- Created a <p> tag to display the validation message after the username input field.
- Conditionally rendered the validation message based on the validation state of the username field.
- Styled the validation message to match the application design.

usama-sattar commented on enatega-multivendor-admin/src/components/Rider/Rider.jsx:
```diff
@@ -223,6 +223,19 @@ function Rider(props) {
+                    marginTop: '5px',
+                    marginLeft: '10px'
+                  }}>
+                  {t('Username cannot contain spaces')}
```
Instead of writing a hard-coded message, make it dynamic and display the error we are actually getting from usernameError. Right now, if the user does not enter anything in the username field, it will still show "space not allowed" error instead of showing username cannot be empty or simply red border.

Also try to make key in translation files for the error message

Nishanth-sebastin replied:
sure i will do it !

usama-sattar commented on enatega-multivendor-admin/src/constraints/constraints.js:
```diff
@@ -74,6 +74,10 @@ const constraints = {
   username: {
     presence: {
       allowEmpty: false
```
You can add a message here as well in allowEmpty check and then use the message based on whatever check is failed to print in the above component.

Nishanth-sebastin commented:
@usama-sattar I have changed it..here you can see it.. https://www.loom.com/share/e7d6f36a765c461eb20210b82ce4b7f1?sid=99acc02d-fff2-425e-a147-1e123dd93f98

I suggest we improve our field validation approach by including both the validation status (isValid) and an error message in the constraint file. This will make future enhancements easier to implement. I've already added this feature to the constraints file for ride js file only without affecting any other files. Let's continue to make our codebase more professional by adopting this standardized approach.

Nishanth-sebastin commented:
Hello team, I just wanted to kindly remind you about my pull request #234  which has been pending review for the past two days. I would appreciate it if someone could take a moment to review and provide feedback. Thank you!

usama-sattar commented on enatega-multivendor-admin/src/components/Rider/Rider.jsx:
```diff
@@ -223,6 +230,20 @@ function Rider(props) {
                     : ''
                 ]}
               />
+              {usernameError === false && (
```
usernameError will be an object return from the validate function. So, this statement will not get rendered even if the error exists.
Initially it will be null and then it will have object of {isValid, errorMessage}. Please tackle null and object case. usernameError === false does not make sense here.

Nishanth-sebastin replied:
The usernameError variable is a simple state used for onSubmit validation. Its only purpose is to indicate whether an error exists (true) or not (false). Throughout the code, you'll notice that most error handling conditions involve checking whether that particular state is true or false. I have set the only isValid property to the usernameError not message so it won't be a object.

Nishanth-sebastin replied:
You're welcome, and I'm excited to contribute more in the future.
